projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts

```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
+            const lastRowSecondCell = getCurrentCellFromGrid(grid, grid.data.length - 1, 1);
+            expectedResult = 'Connor';
+            expect(getValueFromCellElement(lastRowSecondCell)).toEqual(expectedResult);
 
-        grid.rowList.map((item, index) =>
-            expect(grid.getCellByColumn(index, 'ID').value).toEqual(gridData[index].ID));
-    });
+        });
 
-    it('Should have a valid sorting icon when sorting using the API.', () => {
-        const sortingIcon = fixture.debugElement.query(By.css('.sort-icon'));
-        expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
+        it('Should have a valid sorting icon when sorting using the API.', () => {
+            const firstHeaderCell = GridFunctions.getHeaderCellByIndex(fixture, 0);
+            const sortingIcon = GridFunctions.getHeaderSortIcon(firstHeaderCell);
 
-        grid.sort({ fieldName: 'ID', dir: SortingDirection.Asc, ignoreCase: true});
-        fixture.detectChanges();
-        expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
+            expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
 
-        grid.sort({ fieldName: 'ID', dir: SortingDirection.Desc, ignoreCase: true});
-        fixture.detectChanges();
-        expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_DESC_CONTENT);
+            grid.sort({ fieldName: 'ID', dir: SortingDirection.Asc, ignoreCase: true });
+            fixture.detectChanges();
+            expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
 
-        grid.clearSort();
-        fixture.detectChanges();
-        expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
-    });
+            grid.sort({ fieldName: 'ID', dir: SortingDirection.Desc, ignoreCase: true });
+            fixture.detectChanges();
+            expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_DESC_CONTENT);
 
-    it('Should sort grid by clicking on sorting icon when FilterRow is visible.', fakeAsync(/** Filtering showHideArrowButtons RAF */() => {
-        grid.allowFiltering = true;
-        fixture.detectChanges();
+            grid.clearSort();
+            fixture.detectChanges();
+            expect(sortingIcon.nativeElement.textContent.trim()).toEqual(SORTING_ICON_ASC_CONTENT);
+        });
 
-        const initialChips = fixture.debugElement.queryAll(By.directive(IgxChipComponent));
+        it('Should sort grid on sorting icon click when FilterRow is visible.', fakeAsync(/** Filtering showHideArrowButtons RAF */() => {
+            grid.allowFiltering = true;
+            fixture.detectChanges();
 
-        initialChips[0].nativeElement.click();
-        fixture.detectChanges();
+            const fistChip = ControlsFunction.getAllChipsFromAnElement(fixture.debugElement)[0];
```

Comment:
You can change the:
  const fistChip = ControlsFunction.getAllChipsFromAnElement(fixture.debugElement)[0];

            UIInteractions.clickElement(fistChip);
            fixture.detectChanges();

            const filteringRow = fixture.debugElement.query(By.directive(IgxGridFilteringRowComponent));
            expect(filteringRow).toBeDefined();
to:
GridFunctions.clickFilterCellChipUI(fixture, 'Name');
expect(GridFunctions.getFilterRow(fixture)).toBeDefined();